### ISSUE TITLE
SC 4.1.1 also for invisible content?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#1805]
Should 4.1.1 also be checked for invisible content (e.g. display:none, hidden). These are not sent to the Accessibility API, so source code errors should not matter.

**Comment by scottaohara** [comment]
based on a real world use case:

```html
<nameable-element aria-labelledby=foo>...</nameable-element>

<p hidden id=foo>name 1</p>
<div hidden id=foo>oops name</div>
```
That'd still pose an issue though the hidden elements are not directly exposed.

For other sorts of parsing errors, it'd depend if the content _could_ become visible/enabled.  

Some analytical thinking is likely needed in the skillset of the auditor.

**Comment by mraccess77** [comment]
For some things like aria-labelledby which can reference display:none content yes - if you have 2 ids that are the same but different content it could make a difference.  For other things like structure that is never shown it may not make an impact if the error doesn't affect tags that are not display none.

**Comment by JAWS-test** [comment]
@scottaohara I agree with you that in such cases the correct syntax is relevant. However, there are also pages with syntax errors in invisible areas where there is no reference to them via ARIA. But the SC 4.1.1 applies to all content. After all, no one in the audit probably takes the trouble to check each syntax error individually. My experience is that the HTML validator is used. One more reason to [deprecate 4.1.1 ](https://github.com/w3c/wcag/issues/770)

**Comment by scottaohara** [comment]
Yes we do agree on that aspect, @JAWS-test, but your issue here could be interpreted as a flat out ignoring of content that isn't directly exposed to the a11y tree, which could still be problematic as I demonstrated.  

A common 'failure' i see auditors make is failing documents that have `<style>` as a descendant of the `<body>` element. This I think fits very well into for what I think is the intent for this issue.  By the normative language of the SC "elements are nested according to their specifications" then because style is not allowed as a descendant of body this "fails".  But in reality, browsers handle this just fine and there are zero reasons, aside from being pedantic, to fail someone for this "parsing" issue.

I'd very much like to see the understanding document updated to better clarify that HTML validation issues do not always mean parsing issues, and I think this specific issue is a good starting point of where there needs to be nuance to what's written.

**Comment by JAWS-test** [comment]
@scottaohara I don't agree that a style element in the body would be an example that shouldn't cause a violation of SC 4.1.1 because browsers don't have a problem with it. Because that's what 4.1.1 is about: that browsers have repair mechanisms for syntax errors (this is true for most syntax errors checked in 4.1.1, by the way), but first, you can't rely on all browsers and AT to use the same repair mechanisms, and second, there's no guarantee that the mechanisms will work in old or future versions of browsers and AT

> Since repair techniques vary among user agents, authors cannot assume that content will be accurately parsed into a data structure or that it will be rendered correctly by specialized user agents, including assistive technologies, unless the content is created according to the rules defined in the formal grammar for that technology (https://w3c.github.io/wcag/understanding/parsing.html)

> Maximize compatibility with current and future user agents, including assistive technologies. (https://w3c.github.io/wcag/guidelines/#compatible)

My example with the invisible content (which is not referred to by ARIA and which is not switched visible and invisible by scripting) behaves differently: Because these never become relevant for the users of the page, no matter whether with or without AT. Even these must be syntactically correct according to 4.1.1. Therefore my question, whether a corresponding exception should be formulated in the understanding or we continue to stubbornly evaluate every syntax error as a violation of 4.1.1, even if it definitely has no effect on accessibility.

**Comment by scottaohara** [comment]
So then the end goal here is to fail people for some hidden content which isn’t exposed to the accessibility api, but fail them for other content which isn’t exposed to the accessibility api. 

`Style` Of course likely not to have its default UA CSS of display none undone. But an element set to display none by the author could (by a function of the page or a user overwriting the style sheet).  Or, speaking of making sure old browsers are taken into account, guess that means `hidden` Isn’t safe either?
